### PR TITLE
Move incoming webhooks tables over to new schema migrations

### PR DESCRIPTION
**Pull request by nronas**
#### Summary
implements: https://mattermost.atlassian.net/browse/MM-33071

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33071

#### Release Note

```release-note
* Moves Incoming Webhooks gorp driven migrations to schema-based migrations
```

### CODE REVIEW

File: db/migrations/mysql/000006_create_incoming_webhooks.up.sql [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    CreateAt bigint(20) DEFAULT NULL,
+    UpdateAt bigint(20) DEFAULT NULL,
+    DeleteAt bigint(20) DEFAULT NULL,
+    UserId varchar(26) DEFAULT NULL,
+    ChannelId varchar(26) DEFAULT NULL,
+    TeamId varchar(26) DEFAULT NULL,
+    DisplayName varchar(64) DEFAULT NULL,
+    Description text,
+    Username varchar(255) DEFAULT NULL,
+    IconURL text,
```
Comment (mgdelacroix) [inline db/migrations/mysql/000006_create_incoming_webhooks.up.sql:12]: Shouldn't these two be out of the CREATE as we're adding them in the ALTERs?

File: db/migrations/mysql/000006_create_incoming_webhooks.up.sql [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        AND table_schema = DATABASE()
+        AND column_name = 'Description'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE IncomingWebhooks ADD Description text;'
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
```
Comment (mgdelacroix) [inline db/migrations/mysql/000006_create_incoming_webhooks.up.sql:74]: I can't find an ALTER in `upgrade.go` that adds this column, but I've found this one that changes its type:

```go
sqlStore.AlterColumnTypeIfExists("IncomingWebhooks", "Description", "varchar(500)", "varchar(500)")
```

Shouldn't we be changing the type here instead?

File: db/migrations/mysql/000007_create_outgoing_webhooks.up.sql [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        AND table_schema = DATABASE()
+        AND column_name = 'Description'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE OutgoingWebhooks ADD Description text;'
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
```
Comment (mgdelacroix) [inline db/migrations/mysql/000007_create_outgoing_webhooks.up.sql:89]: Same for this description

File: db/migrations/postgres/000007_create_outgoing_webhooks.up.sql [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ALTER TABLE outgoingwebhooks ADD COLUMN IF NOT EXISTS triggerwhen integer default 0;
+ALTER TABLE outgoingwebhooks ADD COLUMN IF NOT EXISTS username VARCHAR(64);
+ALTER TABLE outgoingwebhooks ADD COLUMN IF NOT EXISTS iconurl VARCHAR(1024);
+ALTER TABLE outgoingwebhooks ADD COLUMN IF NOT EXISTS description VARCHAR(500);
+
+CREATE INDEX IF NOT EXISTS idx_outgoing_webhook_team_id ON outgoingwebhooks (teamid);
+CREATE INDEX IF NOT EXISTS idx_outgoing_webhook_update_at ON outgoingwebhooks (updateat);
+CREATE INDEX IF NOT EXISTS idx_outgoing_webhook_create_at ON outgoingwebhooks (createat);
+CREATE INDEX IF NOT EXISTS idx_outgoing_webhook_delete_at ON outgoingwebhooks (deleteat);
+
```
Comment (mgdelacroix) [inline db/migrations/postgres/000007_create_outgoing_webhooks.up.sql:25]:
```suggestion
```

File: db/migrations/mysql/000007_create_incoming_webhooks.up.sql [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    CreateAt bigint(20) DEFAULT NULL,
+    UpdateAt bigint(20) DEFAULT NULL,
+    DeleteAt bigint(20) DEFAULT NULL,
+    UserId varchar(26) DEFAULT NULL,
+    ChannelId varchar(26) DEFAULT NULL,
+    TeamId varchar(26) DEFAULT NULL,
+    DisplayName varchar(64) DEFAULT NULL,
+    Description text,
+    Username varchar(255) DEFAULT NULL,
+    IconURL text,
```
[the remark this answers is not in the document]
Comment (nronas) [reply db/migrations/mysql/000007_create_incoming_webhooks.up.sql]: Same as the earlier PR's trying to mimic what gorp is doing, but i think you are 100% right on this, we prob do not need those in the Creates, will amend

File: db/migrations/mysql/000006_create_incoming_webhooks.up.sql (hunk 2 above)
Comment (nronas) [reply db/migrations/mysql/000006_create_incoming_webhooks.up.sql:74]: 100% copy + 🍝 error

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Pull request merged**